// @semicolondsm/react-emotion-theme/dist/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __markAsModule = (target) => __defProp(target, "__esModule", { value: true });
var __commonJS = (cb, mod) => function __require() {
  return mod || (0, cb[Object.keys(cb)[0]])((mod = { exports: {} }).exports, mod), mod.exports;
};
var __export = (target, all) => {
  __markAsModule(target);
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __reExport = (target, module2, desc) => {
  if (module2 && typeof module2 === "object" || typeof module2 === "function") {
    for (let key of __getOwnPropNames(module2))
      if (!__hasOwnProp.call(target, key) && key !== "default")
        __defProp(target, key, { get: () => module2[key], enumerable: !(desc = __getOwnPropDesc(module2, key)) || desc.enumerable });
  }
  return target;
};
var __toModule = (module2) => {
  return __reExport(__markAsModule(__defProp(module2 != null ? __create(__getProtoOf(module2)) : {}, "default", module2 && module2.__esModule && "default" in module2 ? { get: () => module2.default, enumerable: true } : { value: module2, enumerable: true })), module2);
};

// pnp:/home/runner/work/SemicolonDesign_Frontend/SemicolonDesign_Frontend/packages/design-token/dist/index.js
var require_dist = __commonJS({
  "pnp:/home/runner/work/SemicolonDesign_Frontend/SemicolonDesign_Frontend/packages/design-token/dist/index.js"(exports) {
    var __defProp2 = Object.defineProperty;
    var __markAsModule2 = (target) => __defProp2(target, "__esModule", { value: true });
    var __export2 = (target, all) => {
      __markAsModule2(target);
      for (var name in all)
        __defProp2(target, name, { get: all[name], enumerable: true });
    };
    __export2(exports, {
      colors: () => colors_exports,
      fonts: () => fonts_exports
    });
    var colors_exports = {};
    __export2(colors_exports, {
      auto: () => auto,
      dark: () => dark,
      light: () => light
    });
    var auto = {
      scheme: {
        white: "var(--white)",
        black: "var(--black)",
        purple50: "var(--purple50)",
        purple100: "var(--purple100)",
        purple200: "var(--purple200)",
        purple300: "var(--purple300)",
        purple400: "var(--purple400)",
        purple500: "var(--purple500)",
        purple600: "var(--purple600)",
        purple700: "var(--purple700)",
        purple800: "var(--purple800)",
        purple900: "var(--purple900)",
        gray50: "var(--gray50)",
        gray100: "var(--gray100)",
        gray200: "var(--gray200)",
        gray300: "var(--gray300)",
        gray400: "var(--gray400)",
        gray500: "var(--gray500)",
        gray600: "var(--gray600)",
        gray700: "var(--gray700)",
        gray800: "var(--gray800)",
        gray900: "var(--gray900)",
        yellow50: "var(--yellow50)",
        yellow100: "var(--yellow100)",
        yellow200: "var(--yellow200)",
        yellow300: "var(--yellow300)",
        yellow400: "var(--yellow400)",
        yellow500: "var(--yellow500)",
        yellow600: "var(--yellow600)",
        yellow700: "var(--yellow700)",
        yellow800: "var(--yellow800)",
        yellow900: "var(--yellow900)",
        indigo50: "var(--indigo50)",
        indigo100: "var(--indigo100)",
        indigo200: "var(--indigo200)",
        indigo300: "var(--indigo300)",
        indigo400: "var(--indigo400)",
        indigo500: "var(--indigo500)",
        indigo600: "var(--indigo600)",
        indigo700: "var(--indigo700)",
        indigo800: "var(--indigo800)",
        indigo900: "var(--indigo900)",
        red50: "var(--red50)",
        red100: "var(--red100)",
        red200: "var(--red200)",
        red300: "var(--red300)",
        red400: "var(--red400)",
        red500: "var(--red500)",
        red600: "var(--red600)",
        red700: "var(--red700)",
        red800: "var(--red800)",
        red900: "var(--red900)",
        green50: "var(--green50)",
        green100: "var(--green100)",
        green200: "var(--green200)",
        green300: "var(--green300)",
        green400: "var(--green400)",
        green500: "var(--green500)",
        green600: "var(--green600)",
        green700: "var(--green700)",
        green800: "var(--green800)",
        green900: "var(--green900)",
        blue50: "var(--blue50)",
        blue100: "var(--blue100)",
        blue200: "var(--blue200)",
        blue300: "var(--blue300)",
        blue400: "var(--blue400)",
        blue500: "var(--blue500)",
        blue600: "var(--blue600)",
        blue700: "var(--blue700)",
        blue800: "var(--blue800)",
        blue900: "var(--blue900)"
      }
    };
    var light = {
      scheme: {
        white: "#FFFFFF",
        black: "#000000",
        purple50: "#F0E6FF",
        purple100: "#D3B3FF",
        purple200: "#B885FF",
        purple300: "#9F62F5",
        purple400: "#9650FA",
        purple500: "#8335F0",
        purple600: "#6D1BE0",
        purple700: "#5A10C2",
        purple800: "#500EAD",
        purple900: "#420399",
        gray50: "#F2F0F5",
        gray100: "#F5F5F5",
        gray200: "#EAE7EE",
        gray300: "#DBD7E0",
        gray400: "#B7B4BD",
        gray500: "#98959E",
        gray600: "#706D75",
        gray700: "#5C5961",
        gray800: "#3F3C42",
        gray900: "#1F1E21",
        yellow50: "#FFFEE6",
        yellow100: "#FFFCB3",
        yellow200: "#FFFB85",
        yellow300: "#FAF673",
        yellow400: "#FAF450",
        yellow500: "#F0E935",
        yellow600: "#E0DA1B",
        yellow700: "#C2BB10",
        yellow800: "#ADA80E",
        yellow900: "#999403",
        indigo50: "#EAE6FF",
        indigo100: "#C0B3FF",
        indigo200: "#9985FF",
        indigo300: "#7A62F5",
        indigo400: "#684DF0",
        indigo500: "#5133E6",
        indigo600: "#391AD6",
        indigo700: "#2B0FB8",
        indigo800: "#260DA3",
        indigo900: "#1A038E",
        red50: "#FFE6E6",
        red100: "#FFB3B5",
        red200: "#FF8589",
        red300: "#F56267",
        red400: "#F04D51",
        red500: "#E63338",
        red600: "#D61A20",
        red700: "#B80F13",
        red800: "#A30D11",
        red900: "#8E0306",
        green50: "#E9FFE6",
        green100: "#BBFFB3",
        green200: "#91FF85",
        green300: "#71F562",
        green400: "#5DF04D",
        green500: "#44E633",
        green600: "#2DD61A",
        green700: "#20B80F",
        green800: "#1CA30D",
        green900: "#118E03",
        blue50: "#E6F2FF",
        blue100: "#B3D6FF",
        blue200: "#85BEFF",
        blue300: "#62A7F5",
        blue400: "#4D99F0",
        blue500: "#3386E6",
        blue600: "#1A72D6",
        blue700: "#0F5EB8",
        blue800: "#0D53A3",
        blue900: "#03448E"
      }
    };
    var dark = {
      scheme: {
        white: "#212121",
        black: "#FFFFFF",
        purple50: "#F0E6FF",
        purple100: "#D3B3FF",
        purple200: "#B885FF",
        purple300: "#9F62F5",
        purple400: "#9650FA",
        purple500: "#8335F0",
        purple600: "#6D1BE0",
        purple700: "#5A10C2",
        purple800: "#500EAD",
        purple900: "#420399",
        gray50: "#FAFAFA",
        gray100: "#F5F5F5",
        gray200: "#EEEEEE",
        gray300: "#E0E0E0",
        gray400: "#BDBDBD",
        gray500: "#9E9E9E",
        gray600: "#757575",
        gray700: "#616161",
        gray800: "#424242",
        gray900: "#FFFFFF",
        yellow50: "#FFFEE6",
        yellow100: "#FFFCB3",
        yellow200: "#FFFB85",
        yellow300: "#FAF673",
        yellow400: "#FAF450",
        yellow500: "#F0E935",
        yellow600: "#E0DA1B",
        yellow700: "#C2BB10",
        yellow800: "#ADA80E",
        yellow900: "#999403",
        indigo50: "#EAE6FF",
        indigo100: "#C0B3FF",
        indigo200: "#9985FF",
        indigo300: "#7A62F5",
        indigo400: "#684DF0",
        indigo500: "#5133E6",
        indigo600: "#391AD6",
        indigo700: "#2B0FB8",
        indigo800: "#260DA3",
        indigo900: "#1A038E",
        red50: "#FFE6E6",
        red100: "#FFB3B5",
        red200: "#FF8589",
        red300: "#F56267",
        red400: "#F04D51",
        red500: "#E63338",
        red600: "#D61A20",
        red700: "#B80F13",
        red800: "#A30D11",
        red900: "#8E0306",
        green50: "#E9FFE6",
        green100: "#BBFFB3",
        green200: "#91FF85",
        green300: "#71F562",
        green400: "#5DF04D",
        green500: "#44E633",
        green600: "#2DD61A",
        green700: "#20B80F",
        green800: "#1CA30D",
        green900: "#118E03",
        blue50: "#E6F2FF",
        blue100: "#B3D6FF",
        blue200: "#85BEFF",
        blue300: "#62A7F5",
        blue400: "#4D99F0",
        blue500: "#3386E6",
        blue600: "#1A72D6",
        blue700: "#0F5EB8",
        blue800: "#0D53A3",
        blue900: "#03448E"
      }
    };
    var fonts_exports = {};
    __export2(fonts_exports, {
      weight: () => weight
    });
    var weight = {
      light: "200",
      regular: "300",
      medium: "400",
      bold: "500"
    };
  }
});

// pnp:/home/runner/work/SemicolonDesign_Frontend/SemicolonDesign_Frontend/packages/react-theming/dist/index.js
var require_dist2 = __commonJS({
  "pnp:/home/runner/work/SemicolonDesign_Frontend/SemicolonDesign_Frontend/packages/react-theming/dist/index.js"(exports) {
    var __defProp2 = Object.defineProperty;
    var __markAsModule2 = (target) => __defProp2(target, "__esModule", { value: true });
    var __export2 = (target, all) => {
      __markAsModule2(target);
      for (var name in all)
        __defProp2(target, name, { get: all[name], enumerable: true });
    };
    __export2(exports, {
      getThemeName: () => getThemeName2
    });
    var getThemeName2 = (mode) => {
      return {
        "auto": "auto",
        "light-only": "light",
        "dark-only": "dark"
      }[mode];
    };
  }
});

// pnp:/home/runner/work/SemicolonDesign_Frontend/SemicolonDesign_Frontend/packages/react-emotion-theme/src/index.ts
__export(exports, {
  SDSThemeProvider: () => SDSThemeProvider
});

// pnp:/home/runner/work/SemicolonDesign_Frontend/SemicolonDesign_Frontend/packages/react-emotion-theme/src/SDSThemeProvider.tsx
var import_react = __toModule(require("react"));
var import_react2 = __toModule(require("@emotion/react"));
var import_design_token = __toModule(require_dist());
var import_react_theming = __toModule(require_dist2());
var SDSThemeProvider = ({ children, mode = "auto" }) => {
  const theme = import_react.default.useMemo(() => {
    const colorTheme = import_design_token.colors[(0, import_react_theming.getThemeName)(mode)];
    return {
      colors: {
        ...colorTheme.scheme
      },
      fonts: import_design_token.fonts
    };
  }, [mode]);
  return /* @__PURE__ */ import_react.default.createElement(import_react2.ThemeProvider, {
    theme
  }, children);
};
